test/e2e/validation_test.go

```diff
@@ -393,3 +395,12 @@ func (tc *testContext) findNodeCSRs(nodeName string) ([]certificates.Certificate
 	}
 	return nodeCSRs, nil
 }
+
+// validateUpgradeableCondition ensures that the operator's Upgradeable condition is correctly communicated to OLM
+func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatus) error {
+	if condition.OcName == "" {
+		// Implies operator is not OLM-managed
+		return nil
+	}
+	return condition.Wait(tc.client.Cache, tc.namespace, operators.Upgradeable, expected)
```

Comment:
Not a huge fan of using product code in the e2e test when the method is not being unit tested.